src/darwin/Framework/CHIP/MTRDevice_XPC.mm

```diff
@@ -293,13 +308,21 @@ - (oneway void)device:(NSNumber *)nodeID internalStateUpdated:(NSDictionary *)di
         return;
     }
 
+    [self _updateInternalState:[dictionary mutableCopy]];
+}
+
+- (void)_updateInternalState:(NSMutableDictionary *)newState
+{
+    VerifyOrReturn([self _ensureValidValuesForKeys:requiredInternalStateKeys inInternalState:newState valueRequired:YES]);
+    VerifyOrReturn([self _ensureValidValuesForKeys:requiredInternalStateKeys inInternalState:newState valueRequired:NO]);
```

Comment:
```suggestion
    VerifyOrReturn([self _ensureValidValuesForKeys:optionalInternalStateKeys inInternalState:newState valueRequired:NO]);
```

Should this line be passing the optional keys?

Reply:
Yes!  Good catch... We clearly need better tests here.